### PR TITLE
feat(z-index): add sass map and utilities - INNO-652

### PR DESCRIPTION
**Pull request by emeryro**
z-indexes are now defined in a sass variable, and used in components.
Current map of z-indexes can be updated if new needs are detected.

Components already upgraded using z-index have been updated.

Note: could not run visual tests for now, I will add them later, but code review can already be done.

### CODE REVIEW

File: framework/components/ecl-dropdowns/ecl-dropdowns.scss [hunk 1]
```diff
@@ -15,5 +15,5 @@
   box-shadow: 0 6px 12px rgba(0, 0, 0, 0.175);
   list-style: none;
   position: absolute;
-  z-index: 1000;
+  z-index: map-get($ecl-z-index, 'auto');
```
Comment (yhuard) [inline framework/components/ecl-dropdowns/ecl-dropdowns.scss]: Why don't you use the `dropdown` index?

File: framework/components/ecl-datepickers/_datepickers.scss [hunk 2]
```diff
@@ -40,7 +40,7 @@ $ecl-datepickers-week-color: #999 !default;
   color: $ecl-datepickers-text-color;
   display: block;
   position: relative;
-  z-index: 9999;
+  z-index: map-get($ecl-z-index, 'modal');
```
Comment (yhuard) [inline framework/components/ecl-datepickers/_datepickers.scss]: Isn't the datepicker more like a dropdown?

File: framework/components/ecl-dropdowns/ecl-dropdowns.scss (hunk 1 above)
Comment (emeryro) [inline framework/components/ecl-dropdowns/ecl-dropdowns.scss]: Yes, my bad

File: framework/components/ecl-datepickers/_datepickers.scss (hunk 2 above)
Comment (emeryro) [inline framework/components/ecl-datepickers/_datepickers.scss]: To me it is more like a popup (= modal).
It has to be on top of other elements (including dropdowns)

**Review by yhuard — APPROVED** [review]

**Review by yhuard — CHANGES_REQUESTED** [review]
Just waiting for the small fix we discussed

**Comment by emeryro** [comment]
Done

**Review by yhuard — APPROVED** [review]

**Comment by yhuard** [comment]
We have to figure out why GH doesn't run the tests nor updates the statuses of the checks...

**Pull request merged**